src/rust/engine/fs/store/src/lib.rs

```diff
@@ -738,6 +768,18 @@ impl Store {
         Some((Err(e), _metadata)) => Err(e),
         None => Err(format!("File with digest {:?} not found", digest)),
       })
+      .inspect(move |metadata: &LoadMetadata| {
```

Comment:
@illicitonion , @cattibrie : As we continue to annotate things, we should probably try to budget some time to look at https://github.com/tokio-rs/tracing (or something related) to see whether there is possible alignment in the future.